launchflow/gcp/gke.py

```diff
@@ -0,0 +1,250 @@
+
+    ```python
+    import launchflow as lf
+
+    cluster1 = lf.gcp.GKECluster("my-cluster1", subnet_ip_cidr_range="10.90.0.0/20")
+    cluster2 = lf.gcp.GKECluster("my-cluster2", subnet_ip_cidr_range="10.50.0.0/20")
+    """
+
+    product = ResourceProduct.GCP_GKE_CLUSTER.value
+
+    def __init__(
+        self,
+        name: str,
+        # TODO: we should add the option to have this auto discovered, currently you will have to change this
+        # if you want to have multiple clusters in the same environment
+        subnet_ip_cidr_range: str = "10.60.0.0/20",
+        pod_ip_cidr_range: str = "192.168.0.0/18",
+        service_cidr_range: str = "192.168.64.0/18",
+        regional: Optional[bool] = None,
+        region: Optional[str] = None,
+        zones: Optional[Union[List[str], str]] = None,
+        delete_protection: bool = False,
+    ):
+        """Creates a new GKE Cluster.
+
+        **Args:**
+        - `name (str)`: The name of the GKE cluster.
+        - `subnet_ip_cidr_range (str)`: The IP range for the cluster's subnet.
+        - `pod_ip_cidr_range (str)`: The IP range for the cluster's pods.
+        - `service_cidr_range (str)`: The IP range for the cluster's services.
+        - `regional (Optional[bool])`: Whether the cluster is regional or zonal. If not provided will default to True for production environments and False for development environments.
+        - `region (Optional[str])`: The region for the cluster. If not provided will default to the default region for the environment.
+        - `zones (Optional[List[str]])`: The zones for the cluster. If not provided will default to the default zone for development environments, and remain unset for production environments.
+        - `delete_protection (bool)`: Whether the cluster should have delete protection enabled.
+        """
+        super().__init__(
+            name,
+            replacement_arguments={
+                "subnet_ip_cidr_range",
+                "pod_ip_cidr_range",
+                "service_cidr_range",
+                "regional",
+                "region",
+                "zones",
+            },
+        )
+        self.subnet_ip_cidr_range = subnet_ip_cidr_range
+        self.pod_ip_cidr_range = pod_ip_cidr_range
+        self.service_cidr_range = service_cidr_range
+        self.regional = regional
+        self.region = region
+        self.zones = None
+        if zones is not None and not isinstance(zones, list):
+            self.zones = [zones]
+        else:
+            self.zones = zones
+        self.delete_protection = delete_protection
+
+    def inputs(self, environment_state: EnvironmentState) -> GKEInputs:
+        zones = self.zones
+        region = self.region or environment_state.gcp_config.default_region  # type: ignore
+        regional = self.regional
+        if not regional:
+            if environment_state.environment_type == EnvironmentType.DEVELOPMENT:
+                regional = False
```

Comment:
does regional mean it deploys to multiple regions? Or just 1?

Reply:
it deploys to multiple zones in a region